crates/oxc_language_server/src/linter/isolated_lint_handler.rs

```diff
@@ -45,12 +45,17 @@ impl IsolatedLintHandlerFileSystem {
 }
 
 impl RuntimeFileSystem for IsolatedLintHandlerFileSystem {
-    fn read_to_string(&self, path: &Path) -> Result<String, std::io::Error> {
+    fn read_to_arena_str<'a>(
+        &self,
+        path: &Path,
+        allocator: &'a Allocator,
+    ) -> Result<&'a str, std::io::Error> {
         if path == self.path_to_lint {
-            return Ok(self.source_text.clone());
+            // TODO: i think we can avoid allocating here.
+            return Ok(allocator.alloc_str(&self.source_text));
```

Comment:
I guess `IsolatedLintHandlerFileSystem` also needs to store source text in an allocator. But I don't know what `IsolatedLintHandlerFileSystem` is, so hard to say! I'm also unclear why the `String` was being cloned previously.

Reply:
from what i understand we get a `String` from the editor (that's using the LSP), and we store it in `IsolatedLintHandlerSystem` so that we can run linter on the edited source text and NOT the text currently on the disk.

So actually i don't think we can avoid calling `alloc_str`, as we'll need the source text in the allocator

Reply:
Well this code is no worse than what was there before anyway - cloning `String` copies the same amount of data as copying string into arena, and this is 1 less heap allocation.